.github/workflows/deploy.yml

```diff
@@ -0,0 +1,136 @@
+name: Deploy 
+
+on: 
+  push: 
+    branches: 
+      - master 
+  pull_request:
+    branches:  
+      - '**'
+
+env: 
+  EASI_APP_NODE_VERSION: '16.14.0'
+  EASI_APP_GO_VERSION: '1.16.6'
+  IMAGE_TAG: ${{ github.event.pull_request.head.sha || github.sha }}
+  SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
+  DOCKER_BUILDKIT: 1
+ 
+
+jobs:
+  Build:
+    uses: ./.github/workflows/initial_build.yml 
+    secrets:
+      DOCKERHUB_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
+      DOCKERHUB_TOKEN: ${{ secrets.DOCKERHUB_TOKEN }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+      AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+
+
+  Server_Test: 
+    uses: ./.github/workflows/server_test.yml 
+    needs: Build
+    secrets: 
+      SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
+      DOCKERHUB_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
+      DOCKERHUB_TOKEN: ${{ secrets.DOCKERHUB_TOKEN }}
+      AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+      AWS_SES_SOURCE_ARN: ${{ secrets.AWS_SES_SOURCE_ARN }}
+      OKTA_TEST_PASSWORD: ${{ secrets.OKTA_TEST_PASSWORD }}
+      OKTA_TEST_SECRET: ${{ secrets.OKTA_TEST_SECRET }}
+      OKTA_TEST_USERNAME: ${{ secrets.OKTA_TEST_USERNAME }}
+      AWS_ROLE_TO_ASSUME: ${{ secrets.AWS_ROLE_TO_ASSUME }}
+      STORYBOOK_S3_BUCKET_DEV: ${{ secrets.STORYBOOK_S3_BUCKET_DEV }}
+        
+
+  deploy: 
+    strategy:
+      matrix: 
```

Comment:
My understanding of `matrix` is that it will run all matrix jobs in parallel, unless you provide a `max parallel` config ([source](https://docs.github.com/en/actions/using-jobs/using-a-build-matrix-for-your-jobs#defining-the-maximum-number-of-concurrent-jobs-in-a-matrix))

Maybe we could break out the `deploy` job into its own workflow and re-use it here (and just have it take parameters for environment and lambda version?)